 # <p align="center"> Bonjour from Team Metagenomics Group A! :handshake:
  
![Visitor](https://visitor-badge.laobi.icu/badge?page_id=prembioinfo.Metagenomics-A)

![alt text](https://github.com/prembioinfo/Metagenomics-A/blob/main/METAGENOMICS.gif)
Spare some time to read this and make sure to peep into our work design. Trust us, you won't regret it!ğŸ¤”

 
# <p align="center"> Metagenomic data analysis to detect antimicrobial resistance gene
We replicated the galaxy tutorial for the detection of antibiotic resistance: https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html
  
+ **Objective of the work:**
  * Perform Quality control on your reads
  * Assemble the genome
  * Determine the structure of the genome(s)
  * Determine the presence of potential antibiotic resistance genes.
  
Prerequisites: Create your account in Galaxy Europe server. (https://usegalaxy.eu/login)
  
## Step 0 - Preparing data
  
We used 8 samples from an original study (Li, R., M. Xie, N. Dong, D. Lin, X. Yang et al., 2018). Efficient generation of complete sequences of MDR-encoding plasmids by rapid assembly of MinION barcoding sequencing data. In the experiment, 12 MDR plasmid-bearing bacterial strains were selected for plasmid extraction, including E. coli, S. typhimurium, V. parahaemolyticus, and K. pneumoniae. Click here for [Datasets](https://doi.org/10.5281/zenodo.3247504)
  
## Step 1 -  Creating a history and building a list collection for the given datasets by importing those datasets in history

* Open your workspace in Galaxy. Make sure that you have an empty analysis history, Give it a name / Create a new history.
* Click the new-history icon at the top of the history panel. If the new-history is missing: Click on the galaxy-gear icon (History options) on the top of the history panel. Select the option Create New from the menu.
* Download the data on your pc from 
https://zenodo.org/record/3247504#.YRrL_IgzbIV
And upload
Or,
Import via links
Copy the links location:

https://zenodo.org/record/3247504/files/RB01.fasta<br />
https://zenodo.org/record/3247504/files/RB02.fasta<br />
https://zenodo.org/record/3247504/files/RB03.fasta<br />
https://zenodo.org/record/3247504/files/RB04.fasta<br />
https://zenodo.org/record/3247504/files/RB05.fasta<br />
https://zenodo.org/record/3247504/files/RB06.fasta<br />
https://zenodo.org/record/3247504/files/RB07.fasta<br />
https://zenodo.org/record/3247504/files/RB08.fasta

Open the Galaxy Upload Manager (galaxy-upload on the top-right of the tool panel) and Select Paste/Fetch Data and copy those above link to retreive input data.<br />
(Tips: Creating a dataset collection - Click on Operations on multiple datasets (check box icon) at the top of the history panel > Check all the datasets in your history you would like to include > Click For all selected and choose Build dataset list > Enter a name for your collection > Click Create List to build your collection > Click on the checkmark icon at the top of your history again.)

## Step 2 - Quality Control: using NanoPlot tool to explore the datasets
- Search for Nanoplot Tool in the search box present on the left side:<br />
	
```
Set Parameters: Select "multifile modeâ€ to batch
â€œType of the file(s) to work onâ€ to fasta 
â€œfilesâ€ to  The Plasmids dataset collection you had created.
Click Execute.
```
## Step 3 - De-novo Assembly: Pairwise alignment using Map with minimap2 tool
- Search for Map with minimap2 Tool in the search box present on the left side:
```
Set Parameters:
Select 
â€œWill you select a reference genome from your history or use a built-in index?â€ to Use a genome from history and build index
â€œUse the following data collection as the reference sequenceâ€ to Plasmids dataset collection you had created.
â€œSingle or Paired-end readsâ€ to Single
â€œSelect fastq datasetâ€ to The Plasmids dataset collection
â€œSelect a profile of preset optionsâ€ to Oxford Nanopore all-vs-all overlap mapping
In the section Set advanced output options: â€œSelect an output formatâ€ to paf
Click Execute.
```
## Step 4 - Ultrafast de novo assembly: using Miniasm tool
- Search for Miniasm Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œSequence Readsâ€ to The Plasmids dataset collection
â€œPAF fileâ€ to Output Minimap dataset collection created by Minimap2 tool
Click Execute.
```
## Step 5 - Remapping: First convert the Assembly Graph (collection) created previously by Miniasm using GFA to Fasta tool then perform pairwise sequence alignment again using Map with minimap2 tool

- Search for GFA to Fasta Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œInput GFA fileâ€ to the Assembly Graph (collection) created by Miniasm
Click Execute. 
```
- Search for Map with minimap2 Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œWill you select a reference genome from your history or use a built-in index?â€ to Use a genome from history and build index
â€œUse the following dataset as the reference sequenceâ€ to FASTA file output from GFA to Fasta tool (collection)
â€œSingle or Paired-end readsâ€ to single
â€œSelect fastq datasetâ€ to The Plasmids collection
â€œSelect a profile of preset optionsâ€ to PacBio/Oxford Nanopore read to reference mapping (-Hk19)
In the section Set advanced output options:
â€œSelect an output formatâ€ to paf
Click Execute.

```

## Step 6: Ultrafast consensus module using Racon tool
- Search for Racon Tool in the search box present on the left side:
```
Set Parameters:
 Select
â€œSequencesâ€ to The Plasmids dataset collection
â€œOverlapsâ€ to the latest PAF file collection created by Minimap2 tool
â€œTarget sequencesâ€ to the FASTA file collection created by GFA to Fasta tool.
Click Execute.

```
## Step 7: Visualize the assemblies using Bandage image tool
- Search for Bandage image Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œGraphical Fragment Assemblyâ€ to the Assembly graph collection created by Miniasm tool
Click Execute.

```
- Explore galaxy-eye for  the output images.

## Step 8: Optimizing assemblies using Create assemblies with Unicycler tool and again visualize using Bandage image tool
- Search for Create assemblies with Unicycler Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œPaired or Single end dataâ€ to None
â€œSelect long reads. If there are no long reads, leave this emptyâ€ to The Plasmids dataset collection
Click Execute.

```
- Search for Bandage image Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œGraphical Fragment Assemblyâ€ to the Final Assembly Graph collection created by Unicycler tool
Click Execute. 

```
- Explore galaxy-eye for  the output images again.
- Use the Scratchbook (galaxy-scratchbook) to compare the  assemblies for the Bandage tool images for the assemblies: The assembly that you got from running minimap2, miniasm, racon tool (first time you ran bandage) with the assembly obtained with Unicycler tool.
## Step 9: Prediction of plasmid sequences and classes using PlasFlow tool
- Search for PlasFlow Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œSequence Readsâ€ to the Final Assembly collection created by Unicycler
Click Execute.

```

## Step 10: Scanning of genome contigs for antimicrobial resistance genes using staramr tool
- Search for staramr Tool in the search box present on the left side:
```
Set Parameters:
Select
â€œgenomesâ€ to the Final Assembly collection created by Unicycler
Click Execute.

```
## Step 11: Extract more information using CARD database (Comprehensive Antibiotic Resistance Database)
- Use the summary file to search the database to get more information about the obtained antibiotic resistant genes. Hence head towards the CARD database (Comprehensive Antibiotic Resistance Database). 

**Analyze the results.**
  
  
 

Thank you for your patience and time!
Have a great day!ğŸ¤—

![alt text](https://github.com/prembioinfo/Metagenomics-A/blob/main/Flowchart.png)
